dali/image/bmp.cc

```diff
@@ -78,14 +78,15 @@ BmpImage::BmpImage(const uint8_t *encoded_buffer, size_t length, DALIImageType i
 
 Image::Shape BmpImage::PeekShapeImpl(const uint8_t *bmp, size_t length) const {
   DALI_ENFORCE(bmp != nullptr);
+  DALI_ENFORCE(length >= 18);
```

Comment:
Why removing this error checking? It is checking that the length is enough to contain the header size. If that's not true we'd be reading out of bounds in line 83